engine/classes/Elgg/UploadService.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Elgg;
+
+use Elgg\Http\Request;
+use ElggFile;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * WARNING: API IN FLUX. DO NOT USE DIRECTLY.
+ *
+ * Use the elgg_* versions instead.
+ *
+ * @access private
+ * @since 2.3
+ */
+class UploadService {
+
+	/**
+	 * @var Request
+	 */
+	private $request;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Request $request Http request
+	 */
+	public function __construct(Request $request) {
+		$this->request = $request;
+	}
+
+	/**
+	 * Returns an array of uploaded file objects regardless of upload status/errors
+	 *
+	 * @param string $input_name Form input name
+	 * @return UploadedFile[]
+	 */
+	public function getUploadedFiles($input_name) {
+		$files = $this->request->files;
+		if (!$files->has($input_name)) {
+			return false;
+		}
+
+		$uploaded_files = $files->get($input_name);
+		if ($uploaded_files instanceof UploadedFile) {
+			$uploaded_files = array($uploaded_files);
+		}
+
+		$wrapped_files = [];
+		if (empty($uploaded_files)) {
+			return $wrapped_files;
+		}
+
+		foreach ($uploaded_files as $index => $uploaded_file) {
```

Comment:
If you are throwing out `\Elgg\UploadedFile` all of this is no longer needed. Just return `$uploaded_files`